recipes/default.rb

```diff
@@ -29,21 +29,40 @@
   owner node['epipe']['user']
   group node['epipe']['group']
   mode 0750
-   variables({ :ndb_connectstring => node['ndb']['connectstring'],
+  variables({ :ndb_connectstring => node['ndb']['connectstring'],
                :database => "hops",
                :meta_database => "hopsworks",
                :elastic_addr => elastic,
             })
 end
 
+template"#{node['epipe']['base_dir']}/bin/reindex-epipe.sh" do
+  source "reindex-epipe.sh.erb"
+  owner node['epipe']['user']
+  group node['epipe']['group']
+  mode 0750
+  variables({ :ndb_connectstring => node['ndb']['connectstring'],
+               :database => "hops",
+               :meta_database => "hopsworks",
+               :elastic_addr => elastic,
+            })
+  only_if { node['elastic']['projects']['reindex'] == "true" }
+end
+
 template"#{node['epipe']['base_dir']}/bin/stop-epipe.sh" do
   source "stop-epipe.sh.erb"
   owner node['epipe']['user']
   group node['epipe']['group']
   mode 0750
 end
 
-
+#when upgrading, we want to reindex the newly created projects index
+bash 'reindex epipe' do
+  user node['epipe']['user']
+  code <<-EOF
+     #{node['epipe']['base_dir']}/bin/reindex-epipe.sh
+  EOF
+end
```

Comment:
shouldn't you check for node['elastic']['projects']['reindex']  or it will just fail if the file is not there?

Reply:
My mistake, I added the check in the template above. Will fix it, thanks.